js/src/edit-block/components/browser-url.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Returns the Post's Edit URL.
+ *
+ * @param {number} postId Post ID.
+ *
+ * @return {string} Post edit URL.
+ */
+export function getPostEditURL( postId ) {
+	return addQueryArgs( 'post.php', { post: postId, action: 'edit' } );
+}
+
+/**
+ * Returns the post's Trashed URL.
+ *
+ * @param {number} postId    Post ID.
+ * @param {string} postType Post Type.
+ *
+ * @return {string} Post trashed URL.
+ */
+export function getPostTrashedURL( postId, postType ) {
+	return addQueryArgs( 'edit.php', {
+		trashed: 1,
+		post_type: postType,
+		ids: postId,
+	} );
+}
+
+/**
+ * Forked from Gutenberg, with no change.
+ *
+ * @todo delete if Gutenberg ever exports this from the @wordpress/edit-post package.
+ * @see https://github.com/WordPress/gutenberg/blob/78585d6935fee9020017d17383cef597b67c5703/packages/edit-post/src/components/browser-url/index.js
+ */
+export class BrowserURL extends Component {
```

Comment:
Todo: Open a PR to Gutenberg that exports this component from `@wordpress/edit-post`